predict.py

```diff
@@ -0,0 +1,143 @@
+                     train_scores_mean + train_scores_std, alpha=0.1,
+                     color="r")
+    plt.fill_between(train_sizes, test_scores_mean - test_scores_std,
+                     test_scores_mean + test_scores_std, alpha=0.1, color="g")
+    plt.plot(train_sizes, train_scores_mean, 'o-', color="r",
+             label="Training score")
+    plt.plot(train_sizes, test_scores_mean, 'o-', color="g",
+             label="Cross-validation score")
+
+    plt.legend(loc="best")
+    return plt
+
+start_time = time.time()
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--model", choices=["random_forest", "k_neighbors", "logistic_regression"], default=["random_forest"], dest="model")
+parser.add_argument("--train_dataset_size", default=100000, type=int, dest="train_dataset_size")
+parser.add_argument("--test_dataset_size", default=0, type=int, dest="test_dataset_size")
+args = parser.parse_args()
+print "Model used in this run is %s" % colored(args.model, "red")
+print "Train dataset size is %s" % colored(args.train_dataset_size, "red")
+print "Test dataset size is %s" % colored(args.test_dataset_size, "red")
+
+
+
+train = open ('train.csv', 'r')
+train_X = []
+train_y = []
+test_X = []
+test_y = []
+cnt = -1
+for line in train: 
+    cnt += 1
+    if cnt == 0:
+        continue
+    (features, label) = parse(line)
+    if cnt <= args.train_dataset_size:
+        train_X.append(features)
+        train_y.append(label)
+    elif cnt <= args.train_dataset_size + args.test_dataset_size:
+        test_X.append(features)
+        test_y.append(label)
+    else: 
+        break
+
+print "Input/Output time: %.3f" % (time.time() - start_time)
+train_start_time = time.time()
+
+if args.model == "logistic_regression":
+    model = linear_model.LogisticRegression(C=1, solver='lbfgs', 
+                                            multi_class='multinomial', max_iter=100, n_jobs=-1)
+elif args.model == "k_neighbors":
+    model = KNeighborsClassifier(n_neighbors=10, n_jobs=-1)
+else:
+    model = RandomForestClassifier(n_estimators=10, max_depth=20, n_jobs=-1)
+
+model_file = open('model.pkl', 'wb')
+my_model = model.fit(train_X, train_y)
+pickle.dump(my_model, model_file)
+
+predicted_probability_distribution_test = model.predict_proba(test_X)
+predicted_probability_distribution_train = model.predict_proba(train_X)
+
+print "Training + prediction time: %.3f" % (time.time() - train_start_time)
+
+MAP_test = mean_average_precision(predicted_probability_distribution_test, 
+                                  model.classes_, 
+                                  test_y)
+print "MAP on test data: %.3f" % MAP_test
+MAP_train = mean_average_precision(predicted_probability_distribution_train, 
+                                   model.classes_, 
+                                   train_y) 
+print "MAP on train data: %.3f" % MAP_train
+
+# plot_learning_curve(model, args.model, train_X, train_y, ylim=None, 
```

Comment:
Remove learning curve from here for now. All the code that you have wrote for the learning curve is saved in history on github, so don't worry about losing it.